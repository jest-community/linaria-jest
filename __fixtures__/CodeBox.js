import React from 'react'; // eslint-disable-line
import { css, names } from 'linaria';

const theme = {
  backdrop: '#ccc',
  primary: '#123',
  hover: '#fff',
};

export default function CodeBox({ className, text }) {
  return (
    <div className={container}>
      <pre className={names(code, className)}>
        <code className={className}>{text}</code>
      </pre>
    </div>
  );
}

const code = css`
  font-size: 14px;
  padding: 16px;
  background: ${theme.backdrop};
  color: ${theme.primary};
  overflow: auto;
`;

const container = css`
  max-width: 900px;

  &:hover {
    background: ${theme.hover};
  }
`;
