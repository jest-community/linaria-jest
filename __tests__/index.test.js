/* @flow */
import React from 'react';
import renderer from 'react-test-renderer';
import { css } from 'linaria';
import { reactSerializer } from '..';
import CodeBox from '../__fixtures__/CodeBox';

expect.addSnapshotSerializer(reactSerializer);

test('reactSerializer', () => {
  const tree = renderer.create(
    <CodeBox
      text="test react"
      className={css`
        flex: 1;
      `}
    />
  );

  expect(tree).toMatchSnapshot();
});
