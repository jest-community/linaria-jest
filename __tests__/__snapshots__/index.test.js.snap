// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reactSerializer 1`] = `
.a0 {
  font-size: 14px;
  padding: 16px;
  background: #ccc;
  color: #123;
  overflow: auto;
}
.a1 {
  max-width: 900px;
}
.a1:hover {
  background: #fff;
}
.a3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="a1"
>
  <pre
    className="a0 a3"
  >
    <code
      className="a3"
    >
      test react
    </code>
  </pre>
</div>
`;
